test/e2e/framework/util.go

```diff
@@ -1243,53 +1263,50 @@ func RunCmdEnv(env []string, command string, args ...string) (string, string, er
 	return stdout, stderr, nil
 }
 
-// getControlPlaneAddresses returns the externalIP, internalIP and hostname fields of control plane nodes.
-// If any of these is unavailable, empty slices are returned.
-func getControlPlaneAddresses(c clientset.Interface) ([]string, []string, []string) {
-	var externalIPs, internalIPs, hostnames []string
+// getMasterAddresses returns the externalIP, internalIP and hostname fields of the master.
+// If any of these is unavailable, it is set to "".
+func getMasterAddresses(c clientset.Interface) (string, string, string) {
+	var externalIP, internalIP, hostname string
 
-	// Populate the internal IPs.
+	// Populate the internal IP.
 	eps, err := c.CoreV1().Endpoints(metav1.NamespaceDefault).Get(context.TODO(), "kubernetes", metav1.GetOptions{})
 	if err != nil {
 		Failf("Failed to get kubernetes endpoints: %v", err)
 	}
-	for _, subset := range eps.Subsets {
-		for _, address := range subset.Addresses {
-			if address.IP != "" {
-				internalIPs = append(internalIPs, address.IP)
-			}
-		}
+	if len(eps.Subsets) != 1 || len(eps.Subsets[0].Addresses) != 1 {
+		Failf("There are more than 1 endpoints for kubernetes service: %+v", eps)
 	}
+	internalIP = eps.Subsets[0].Addresses[0].IP
```

Comment:
?? 
there can be multiple endpoints

Reply:
This is a bad merge from a branch, i'm fixing it.

Reply:
:smile: